tuna/export_db.py

```diff
@@ -95,237 +105,240 @@ def parse_args():
                      default=False)
   args = parser.parse_args()
 
+  if args.golden_v and not (args.arch and args.num_cu):
+    parser.error('arch and num_cu must be set with golden_v')
+
   return args
 
 
-def fdb_query(dbt, args):
+def get_filename(arch, num_cu, filename, ocl, db_type):
+  """Helper function to compose filename"""
+  version = "1.0.0"
+  tuna_dir = f'tuna_{version}'
+  if not os.path.exists(tuna_dir):
+    os.makedirs(tuna_dir)
+  final_name = f"{tuna_dir}/{arch}_{num_cu}"
+  if num_cu > 64:
+    final_name = f'{tuna_dir}/{arch}{num_cu:x}'
+  if filename:
+    final_name = f'{tuna_dir}/{filename}'
+
+  if db_type == DB_Type.FIND_DB:
+    # pylint: disable-next=consider-using-f-string ; more readable
+    extension = '.{}.fdb.txt'.format('OpenCL' if ocl else 'HIP')
+  elif db_type == DB_Type.KERN_DB:
+    extension = '.kdb'
+  else:
+    extension = ".db"
+
+  final_name = f"{final_name}{extension}"
+
+  return final_name
+
+
+def get_fdb_query(dbt, args):
   """ Helper function to create find db query
   """
-  find_db_table = dbt.find_db_table
-  config_tags_table = dbt.config_tags_table
-  solver_app = dbt.solver_app
-  with DbSession() as session:
-    query = session.query(find_db_table, solver_app).filter(
-        and_(find_db_table.session == dbt.session.id,
-             solver_app.session == dbt.session.id,
-             find_db_table.kernel_time != -1, find_db_table.workspace_sz != -1,
-             find_db_table.opencl == args.opencl, find_db_table.valid == 1))\
-             .filter(find_db_table.solver == solver_app.solver,
-                     find_db_table.config == solver_app.config)\
-             .order_by(find_db_table.config, find_db_table.update_ts.desc())
-
-    LOGGER.info("Collecting %s entries.", find_db_table.__tablename__)
-    query = query.filter(solver_app.applicable == 1)
-    LOGGER.info("rocm_v : %s", dbt.session.rocm_v)
-    LOGGER.info("miopen_v : %s", dbt.session.miopen_v)
-    if args.config_tag:
-      LOGGER.info("config_tag : %s", args.config_tag)
-      tag_query = session.query(config_tags_table.config).filter(
-          config_tags_table.tag == args.config_tag)
-      tag_rows = tag_query.all()
-      ids = tuple((str(tag_row.config) for tag_row in tag_rows))
-      query = query.filter(find_db_table.config.in_(ids))
+  src_table = dbt.find_db_table
+  if args.golden_v is not None:
```

Comment:
when would we have args.golden_v == None? if we require the argument then we can remove this check

Reply:
We require exactly one of session_id and golden_v, golden_v could be None.